x-pack/plugins/observability/server/services/slo/tasks/orphan_summary_cleanup_task.ts

```diff
@@ -0,0 +1,246 @@
+            sloSummaryIds.map(({ id }) => ({
+              id,
+              type: SO_SLO_TYPE,
+            })),
+            {
+              namespace: ALL_SPACES_ID,
+            }
+          )
+        ).saved_objects.filter((so) => !so.error);
+
+        const sloSummaryIdsToDelete = sloSummaryIds.filter(
+          ({ id, revision }) =>
+            !sloDefinitions.find(
+              ({ attributes }) => attributes.id === id && attributes.revision === revision
+            )
+        );
+
+        if (sloSummaryIdsToDelete.length > 0) {
+          this.logger.info(
+            `[SLO] Deleting ${sloSummaryIdsToDelete.length} SLO Summaries from the summary index`
+          );
+
+          await this.esClient.deleteByQuery({
+            wait_for_completion: false,
+            index: SLO_SUMMARY_DESTINATION_INDEX_PATTERN,
+            query: {
+              bool: {
+                should: getDeleteQueryFilter(sloSummaryIdsToDelete.sort()),
+              },
+            },
+          });
+        }
+      } while (searchAfterKey);
+    }
+
+    return { state: { lastRunAt: runAt } };
+  };
+
+  fetchSloSummariesIds = async (
+    searchAfter?: AggregationsCompositeAggregateKey
+  ): Promise<{
+    searchAfter?: AggregationsCompositeAggregateKey;
+    sloSummaryIds: Array<{ id: string; revision: number }>;
+  }> => {
+    this.logger.debug(`[SLO] Fetching SLO Summaries ids after ${searchAfter}`);
+    if (!this.esClient) {
+      return {
+        searchAfter: undefined,
+        sloSummaryIds: [],
+      };
+    }
+
+    const size = 1000;
+
+    const result = await this.esClient.search<
+      unknown,
+      {
+        slos: {
+          after_key: AggregationsCompositeAggregateKey;
+          buckets: Array<{
+            key: {
+              id: string;
+              revision: number;
+            };
+          }>;
+        };
+      }
+    >({
+      size: 0,
+      index: SLO_SUMMARY_DESTINATION_INDEX_PATTERN,
+      aggs: {
+        slos: {
+          composite: {
+            size,
+            sources: [
+              {
+                id: {
+                  terms: {
+                    field: 'slo.id',
+                  },
+                },
+              },
+              {
+                revision: {
+                  terms: {
+                    field: 'slo.revision',
+                  },
+                },
+              },
+            ],
+            after: searchAfter,
+          },
+        },
+      },
+    });
+
+    const aggBuckets = result.aggregations?.slos.buckets ?? [];
+    if (aggBuckets.length === 0) {
+      return {
+        searchAfter: undefined,
+        sloSummaryIds: [],
+      };
+    }
+
+    const newSearchAfter =
+      aggBuckets.length < size ? undefined : result.aggregations?.slos.after_key;
+
+    const sloSummaryIds = aggBuckets.map(({ key }) => {
+      return {
+        id: key.id as string,
+        revision: key.revision as number,
```

Comment:
Let's not only please typescript, but make sure the type is correct at runtime
```suggestion
        id: String(key.id),
        revision: Number(key.revision),
```